File: PYTHON/Algorithms/Sort/2. BubbleSort.py
from ListGeneration import listGenerator
import time

arr5K = listGenerator(5000)

def bubble_sort(arr):
    for i in range(len(arr)):
        sorted = True
        for j in range(len(arr) - 1 - i):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                sorted = False
        if sorted:
            break

# Record the start time
start_time = time.time()

bubble_sort(arr5K)

# Record the end time
end_time = time.time()

# It will take at least 1 seconds to sort the arr5K

# if you want to visualize starting array just comment bubble_sort(arr5K) on previous lines
print(arr5K)
print('----------------------------------------------------------------')
print('// Stats of Bubble Sort //')
print('Length --> ', len(arr5K))
print("Time taken:", end_time - start_time, "seconds")

File: TYPESCRIPT/Algorithms/Sort/2. BubbleSort.ts
import { generateArray } from './ArrayGeneration';

// Generates an array of 25000 random numbers from -20000 to 20000
const arr25K: number[] = generateArray(25000);

const bubbleSort = (arr: number[]) => {
  for (let i: number = 0; i < arr.length; i++) {
    let sorted: boolean = true;
    for (let j: number = 0; j < arr.length - i - 1; j++) {
      if (arr[j] > arr[j + 1]) {
        let temp: number = arr[j];
        arr[j] = arr[j + 1];
        arr[j + 1] = temp;
        sorted = false;
      }
    }
    if (sorted) break;
  }
};

const arr: number[] = [...arr25K];

// Measure execution time
const startTime: number = performance.now();
bubbleSort(arr);
// if you want to visualize starting array just change arr to arr25K
const sortedArrayString: string = arr25K.join(', ');
const endTime: number = performance.now();

// It will take at least 0.7 seconds to sort the arr25K
const timeTakenInSeconds: number = (endTime - startTime) / 1000;

console.log('[', sortedArrayString, ']');
console.log('--------------------------------------------------------');
console.log('// Stats of Bubble Sort //');
console.log('Length --> ', arr.length);
console.log('Time taken:', timeTakenInSeconds, 'seconds');
